dali/operators/generic/join.cc

```diff
@@ -0,0 +1,254 @@
+  .AddOptionalArg<int>("axis", "Axis along which the input tensors are concatenated.", 0, false)
+  .AddOptionalArg<string>("axis_name", R"(Name of the axis along which the tensors are concatenated.
+
+This argument is mutually exclusive with ``axis``.
+This argument requires that at least one input has a non-empty layout and that all non-empty
+input layouts match.)", nullptr, false)
+  .NumInput(1, 999)
+  .NumOutput(1);
+
+DALI_SCHEMA(Stack)
+  .DocStr(R"(Joins the input tensors along a new axis.
+
+The shapes of respective tensors in the inputs must match.)")
+  .AddOptionalArg<int>("axis", R"(The axis in the output tensor along which the inputs are stacked.
+
+The axis is inserted before a corresponding axis in the inputs. A value of 0 indicates that whole
+tensors are stacked. Speicfying ``axis`` equal to the number of dimensions in the inputs causes
+the values from the inputs to be interleaved)", 0, false)
+  .AddOptionalArg<string>("axis_name", R"(Name of the new axis to be inserted.
+
+A one-character that will denot the new axis in the output layout. The output layout will be
+constructed by inserting that character into the input layout at position indicated by ``axis``.
+For example, specifying ``axis = 0`` and ``axis_name = "C"`` with input layout "HW" will yield
+the output layout "CHW")", nullptr, false)
+  .NumInput(1, 999)
+  .NumOutput(1);
+
+#define TENSOR_JOIN_TYPES (bool, uint8_t, int8_t, uint16_t, int16_t, uint32_t, int32_t, \
+                          uint64_t, int64_t, float16, float, double)
+
+template <typename Backend, bool new_axis>
+bool TensorJoin<Backend, new_axis>::SetupImpl(
+        vector<OutputDesc> &outputs, const workspace_t<Backend> &ws) {
+  int njoin = this->spec_.NumRegularInput();
+  outputs.resize(1);
+  outputs[0].type = ws.template InputRef<Backend>(0).type();
+  auto &output_shape = outputs[0].shape;
+
+  // Check that all inputs have the same type
+  DALIDataType out_type = outputs[0].type.id();
+  for (int i = 1; i < njoin; i++) {
+    DALIDataType type_id = ws.template InputRef<Backend>(i).type().id();
+    DALI_ENFORCE(type_id == out_type, make_string(
+        "All inputs must have the same type.\nType of input #0: ", out_type,
+        "\nType of input #", i, ": ", type_id));
+  }
+
+  GetInputLayout(ws);
+  SetupAxis();
+  SetOutputLayout(ws);
+
+  // Run ove inputs and store them in a vector
+  TYPE_SWITCH(out_type, type2id, T, TENSOR_JOIN_TYPES, (
+    SetupTyped<T>(output_shape, ws);
+  ), (DALI_FAIL(make_string("The element type ", out_type, " is not supported."))));  // NOLINT
+  return true;
+}
+
+template <typename Backend, bool new_axis>
+template <typename T>
+void TensorJoin<Backend, new_axis>::SetupTyped(
+      TensorListShape<> &output_shape, const workspace_t<Backend> &ws) {
+  auto &inputs = this->template inputs<T>();
+  inputs.clear();
+  int ninp = this->spec_.NumRegularInput();
+
+  copy_idx_ = 0;
```

Comment:
```suggestion
  copy_only_ = false;
```

Reply:
??? That's a _number_!

Reply:
And I don't think it should be. As I understand it is to tell if it should just copy or not. Making it an int blurs the purpose.

Reply:
But it also tell _which_ one should be copied.